purchase_sale_inter_company/models/purchase_order.py

```diff
@@ -0,0 +1,163 @@
+                purchase_line, dest_company, sale_order)
+            self.env['sale.order.line'].sudo(
+                intercompany_user.id).create(sale_line_data)
+        # write supplier reference field on PO
+        if not self.partner_ref:
+            self.partner_ref = sale_order.name
+        # Validation of sale order
+        if dest_company.sale_auto_validation:
+            sale_order.sudo(intercompany_user.id).action_confirm()
+
+    @api.multi
+    def _prepare_sale_order_data(self, name, partner, dest_company,
+                                 direct_delivery_address):
+        """ Generate the Sale Order values from the PO
+            :param name : the origin client reference
+            :rtype name : string
+            :param partner : the partner reprenseting the company
+            :rtype partner : res.partner record
+            :param dest_company : the company of the created SO
+            :rtype dest_company : res.company record
+            :param direct_delivery_address : the address of the SO
+            :rtype direct_delivery_address : res.partner record
+        """
+        self.ensure_one()
+        delivery_address = (
+            direct_delivery_address or
+            self.picking_type_id.warehouse_id.partner_id or False)
+        new_order = self.env['sale.order'].new({
+            'company_id': dest_company.id,
+            'client_order_ref': name,
+            'partner_id': partner.id,
+            'date_order': self.date_order,
+            'auto_purchase_order_id': self.id,
+        })
+        for onchange_method in new_order._onchange_methods['partner_id']:
+            onchange_method(new_order)
+        new_order.user_id = False
+        if delivery_address:
+            new_order.partner_shipping_id = delivery_address
+        if self.notes:
+            new_order.note = self.notes
+        if 'warehouse_id' in new_order:
+            new_order.warehouse_id = (
+                dest_company.warehouse_id.company_id == dest_company and
+                dest_company.warehouse_id or False)
+        if 'requested_date' in new_order:
+            new_order.requested_date = self.date_planned
+        return new_order._convert_to_write(new_order._cache)
+
+    @api.model
+    def _prepare_sale_order_line_data(
+            self, purchase_line, dest_company, sale_order):
+        """ Generate the Sale Order Line values from the PO line
+            :param purchase_line : the origin Purchase Order Line
+            :rtype purchase_line : purchase.order.line record
+            :param dest_company : the company of the created SO
+            :rtype dest_company : res.company record
+            :param sale_order : the Sale Order
+        """
+        new_line = self.env['sale.order.line'].new({
+            'order_id': sale_order.id,
+            'product_id': purchase_line.product_id.id,
+            'product_uom': purchase_line.product_id.uom_id.id,
```

Comment:
Shouldn't it be `purchase_line.product_uom.id`? 
E.g.: you order 1 dozen of eggs but the sale order registers 1 unit of eggs (default uom in the product itself) and the price in the PO is syncronized to price of that unit. So the result is the company selling serves just 1 egg.

Reply:
Of course. This should be purchase_line.product_uom.id